freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/dns/DnsRecordService.java

```diff
@@ -48,6 +48,18 @@ public void deleteDnsRecordByFqdn(String environmentCrn, String accountId,  List
                 LOGGER.info("Delete DNS record [{}] in zone [{}]", dnsRecord, dnsZone);
                 freeIpaClient.deleteDnsRecord(dnsRecord.getIdnsname(), dnsZone.getIdnsname());
             }
+            Set<DnsRecord> srvRecordsToDelete = allDnsRecordInZone.stream()
+                    .filter(record -> fqdns.stream().anyMatch(fqdn -> record.isHostRelatedSrvRecord(fqdn)))
+                    .collect(Collectors.toSet());
+            for (DnsRecord dnsRecord : srvRecordsToDelete) {
+                for (String fqdn : fqdns) {
+                    List<String> srvRecords = dnsRecord.getHostRelatedSrvRecords(fqdn);
+                    if (!srvRecords.isEmpty()) {
+                        LOGGER.info("Delete DNS SRV record [{}] for [{}] in zone [{}]", dnsRecord.getIdnsname(), fqdn, dnsZone);
+                        freeIpaClient.deleteDnsSrvRecord(dnsRecord.getIdnsname(), dnsZone.getIdnsname(), srvRecords);
+                    }
+                }
+            }
```

Comment:
I think it's worth to refactor the body of the `for (DnsZoneList dnsZone : freeIpaClient.findAllDnsZone()) {` and introduce 1 or 2 new methods